lua/plenary/functional.lua

```diff
@@ -38,4 +38,19 @@ f.all = function(f, iterable)
   return true
 end
 
+function f.first(...)
+  local x = select(1, ...)
+  return x
+end
```

Comment:
Instead of copy and pasting these with just 1, 2, 3, different, can we make this in a different way?

Reply:
Hmm, I'm not really sure. The nice thing about these functions is that you can do things like `first(1, 2, 3)` which is pretty minimal, whereas something like `select_arg(1, 1, 2, 3)` is less so. We could do something like this though:
```lua
local function make_selector(n, ...) -- Need a better name
    return function(...) local x = select(n, ...) return x end
end

f.first = make_selector(1)
f.second = make_selector(2)
-- Etc.
```
Although I'm not sure if that's really better or not. But if we expose `make_selector` or whatever we call it, that would provide the user with an easy way to make more of these sorts of functions if they need to.

Reply:
You don't need the `...` in the make_selector definition.

Yeah, that's what I was thinking though.

Reply:
Also, can you add a `last`?

Reply:
How would you implement `last`? I'm guess it'd have to be using something other than `select`, but I'm not really sure what.

Reply:
Do you want a hint or the answer (I'm using this as a chance to get ahead of future lua questions :wink: ). If that's annoying, then I will just tell you :laughing:

Reply:
Haha, well definitely the hint ;)

My initial thought would be:
```lua
local function last(...)
  local length = select('#', ...)
  local x = select(length, ...)
  return x
end

print(last(5, 6, 7)) -- "7"
```
But that seems kind of inefficient maybe? Granted, I haven't done any benchmarks so that's just a guess that may be wrong.

Reply:
Perfect, that's what I was thinking as well :) (at least for a first shot, we can always optimize later).